### PR TITLE
Nodes, Items and Crafting - Added more crafting types.

### PR DESCRIPTION
**Pull request by Amaz1**
I imagine that there will be typos here, and other things that need correcting, but here is a basic, first version of it!

### CODE REVIEW

File: chapters/nodes_items_crafting.md [hunk 1]
```diff
@@ -227,7 +228,93 @@ minetest.register_craft({
...
+Note: Other options, like the number after the result, found in the shaped
+crafting recipe, can be used here, and for some of the other craft types.
+
+###Cooking
+
+Recipes with the type "cooking" are not made in the crafting grid, like
+other recipes, but as the name suggests, they are cooked, in furnaces,
+or other cooking tools that might be found in mods.
+This is the way that anything that must be cooked, like bricks, or
+can be cook to turn into something else, like cobble, are defined.
```
Comment (ExcaliburZero) [inline chapters/nodes_items_crafting.md]: "can be cook" should be "can be cooked"

File: chapters/nodes_items_crafting.md [hunk 2]
```diff
@@ -227,7 +228,93 @@ minetest.register_craft({
...
+	burntime = 300,
+})
+{% endhighlight %}
+
+They don't have an output, like other recipes, but they have a burn time,
+which defines how long they will last as fuel, in seconds. 
+So, the diamond is good as fuel for 300 seconds!
+
+###Tool Repair
+This is a fairly simple type of craft. It simple allows two tools, of the
```
Comment (ExcaliburZero) [inline chapters/nodes_items_crafting.md]: "It simple allows" should be "It simply allows"

**Comment by Amaz1** [comment]
Fixed both of those, thanks!

File: chapters/nodes_items_crafting.md [hunk 3]
```diff
@@ -227,7 +228,93 @@ minetest.register_craft({
...
+
+###Tool Repair
+This is a fairly simple type of craft. It simply allows two tools, of the
+same type,  to be crafted together, and produce a tool with the
+combined wear of the two separate tools, minus the additional wear
+as defined in the additional_wear field. 
+
+{% highlight lua %}
+minetest.register_craft({
+	type = "toolrepair",
```
Comment (rubenwardy) [inline chapters/nodes_items_crafting.md]: Looks like you're missing some way of specifying which tool you're actually repairing. That's probably quite important.

File: chapters/nodes_items_crafting.md [hunk 4]
```diff
@@ -182,6 +182,7 @@ identified by the ``type`` property.
 * shapeless - It doesn't matter where the ingredients are,
   just that there is the right amount.
 * cooking - Recipes for the furnace to use.
+* -- fuel - Defines what can be burnt in furnaces.
```
Comment (rubenwardy) [inline chapters/nodes_items_crafting.md]: For a sub list, I think you're supposed to tab in like this:

    * one
        * two

File: chapters/nodes_items_crafting.md (hunk 3 above)
Comment (Amaz1) [inline chapters/nodes_items_crafting.md]: As far as I could see from the C++ code (https://github.com/minetest/minetest/blob/master/src/script/lua_api/l_craft.cpp#L213-L220) the only option that is available for the toolrepair is the additional wear field. I may be missing something however...

File: chapters/nodes_items_crafting.md (hunk 4 above)
Comment (Amaz1) [inline chapters/nodes_items_crafting.md]: Fixed, thanks.

File: chapters/nodes_items_crafting.md (hunk 3 above)
Comment (rubenwardy) [inline chapters/nodes_items_crafting.md]: That's just stupid. You are right.

**Comment by rubenwardy** [comment]
I didn't much tool repair, as it would be rather confusing for new modders, IMO.

**Pull request merged**